src/viam/app/app_client.py

```diff
@@ -324,6 +347,27 @@ def proto(self) -> RobotPartHistoryEntryPB:
         )
 
 
+class APIKeyAuthorization:
+    """A class with the necessary authorization data for creating an API key.
+
+    Use this class when constructing API key authorizations to minimize the risk of malformed or missing data.
+    """
+
+    def __init__(self, role: str, resource_type: str, resource_id: str):
+        """role (str): The role to add (i.e., either "owner" or "operator").
+        resource_type (str): Type of the resource to add role to (i.e., either "organization", "location", or "robot").
+        Must match `resource_id`.
+        resource_id (str): ID of the resource the role applies to (i.e., either an organization, location, or robot ID).
+        """
+        self._role = role
+        self._resource_type = resource_type
+        self._resource_id = resource_id
+
+    _role: str
+    _resource_type: str
+    _resource_id: str
```

Comment:
Is there a reason for these being private?

Reply:
In general I favor minimizing direct access to data members. In this case the class is kind of weird/bespoke compromise (it's an opinionated version of the API key authorization information for a single, specific use case) and so I favored minimizing the amount of interaction users have with it. It's not a super strong preference though, happy to revisit if you disagree!